src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_NullChecking.cs

```diff
@@ -62,8 +56,10 @@ private static BoundStatement ConstructIfStatementForParameter(SourceParameterSy
             {
                 paramIsNullCondition = factory.MakeNullCheck(loweredLeft.Syntax, loweredLeft, BinaryOperatorKind.Equal);
             }
-            // PROTOTYPE : Make ArgumentNullException
-            BoundThrowStatement throwArgNullStatement = factory.Throw(factory.New(factory.WellKnownType(WellKnownType.System_Exception)));
+
+            ImmutableArray<BoundExpression> exceptionName = ArrayBuilder<BoundExpression>.GetInstance(1, factory.StringLiteral(parameter.Name)).ToImmutableAndFree();
```

Comment:
>ArrayBuilder<BoundExpression>.GetInstance(1, factory.StringLiteral(parameter.Name)).ToImmutableAndFree(); [](start = 60, length = 105)

is there a reason this is being used instead of `ImmutableArray.Create(factory.StringLiteral(parameter.Name))`?